scripts/keys.coffee

```diff
@@ -10,95 +10,247 @@
 #	bot xyz has keys : sets xyz as the holder of keys
 #
 # Examples:
-#   :bot who has keys
-#   :bot i have keys
-#   :bot i dont have keys
-#	:bot who has keys
-#	:bot ravi has keys
+#   :bot who has <color> keys
+#   :bot i have <color> keys
+#   :bot i dont have <color> keys
+#	:bot who has <color> keys
+#	:bot ravi has <color> keys
 #
 # Author:
 #   Punit Dhoot (@pdhoot)
-#   Developer at SDSLabs (@sdslabs)
+#   Developer at SDSLabs (@sdslabs
+#  Edit:
+#   prakhar gupta
+
 
 module.exports = (robot)->
 
-	key = ()->
-		Key = robot.brain.get("key") or ""
-		robot.brain.set("key" ,Key)
-		Key	
+	keyRed = ()->
+		Key1 = robot.brain.get("keyRed") or ""
+		robot.brain.set("keyRed",Key1)
+		Key1	
+	keyBlue = ()->
+		Key2 = robot.brain.get("keyBlue") or ""
+		robot.brain.set("keyBlue",Key2)
+		Key2	
+	keyYellow = ()->
+		Key3 = robot.brain.get("keyYellow") or ""
+		robot.brain.set("keyYellow",Key3)
+		Key3	
+	keyGreen = ()->
+		Key4 = robot.brain.get("keyGreen") or ""
+		robot.brain.set("keyGreen",Key4)
+		Key4	
+	keyOrange = ()->
+		Key5 = robot.brain.get("keyOrange") or ""
+		robot.brain.set("keyOrange",Key5)
+		Key5	
 
-	
-	robot.respond /i have (a key|the key|key|keys)/i, (msg)->
+    #1this section belongs to "i have keys"
+	robot.respond /i (have the|have) (.+) (key|keys)/i, (msg)->
+		keyname = msg.match[2]
 		name = msg.message.user.name 
 		user = robot.brain.userForName name
-		k = key()
-		if typeof user is 'object'
-			msg.send "Okay #{name} has keys"
-			k = user.name
-		robot.brain.set("key",k)	
-
+		unless keyname not in  ["red", "blue", "yellow", "green", "orange"]
+			if keyname is 'red'
+				key_holder = keyRed()
+				key_holder = user.name
+				robot.brain.set("keyRed",key_holder)
+			else if keyname is 'blue'
+				key_holder = keyBlue()
+				key_holder = user.name
+				robot.brain.set("keyBlue",key_holder)
+			else if keyname is 'Yellow'
+				key_holder = keyYellow()
+				key_holder = user.name
+				robot.brain.set("keyYellow",key_holder)
+			else if keyname is 'Green'
+				key_holder = keyGreen()
+				key_holder = user.name
+				robot.brain.set("keyGreen",key_holder)
+			else if keyname is 'orange'
+				key_holder = keyOrange()
+				key_holder = user.name
+				robot.brain.set("keyOrange",key_holder)
+			if typeof user is 'object'
+				msg.send "Okay #{name} has #{keyname} keys"	
 
-	robot.respond /i (don\'t|dont|do not) (has|have) (the key|key|keys|a key)/i , (msg)->
+	#2this section belongs to "i dont have key"
```

Comment:
Also, create a command for "I don't have keys"
This should check what kind of keys I have, for example: Blue. 
Then it should assign null to the blue keys and prompt asking who has blue keys.

Reply:
ok